### PR TITLE
feat: add plugin-register;

### PR DESCRIPTION
**Pull request by web3max**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -119,6 +119,7 @@ For detailed usage, please follow the instructions in the specific plugin's READ
 - [@dotbit/plugin-template](./packages/plugin-template/README.md): A demo plugin demonstrating the basic structure of a .bit plugin.
 - [@dotbit/plugin-web3mq](./packages/plugin-web3mq/README.md): A plugin for integrating [Web3MQ](https://www.web3messaging.online/).
 - [@dotbit/plugin-avatar](./packages/plugin-avatar/README.md): A plugin for resolving users' avatar.
+- [@dotbit/plugin-register](./packages/plugin-register/README.md): A plugin for register .bit account.
```
Comment (zgayjjf) [inline README.md]: "A plugin to register .bit account" or "A plugin for registering .bit account."

File: src/const.ts [hunk 2]
```diff
@@ -58,7 +58,23 @@ export enum RecordType {
 }
 
 export enum AccountStatus {
-  noneExist,
+  notOpenRegister= -1,
```
Comment (zgayjjf) [inline src/const.ts:61]: It would be great to place a link to the source in backend.

File: src/const.ts [hunk 3]
```diff
@@ -143,3 +159,13 @@ export enum DigitalEmojiUnifiedMap {
   '8⃣' = '8️⃣',
   '9⃣' = '9️⃣'
 }
+
+// ID of the payment method, source: https://register-api.did.id/v1/token/list
```
Comment (zgayjjf) [inline src/const.ts]: 404, add a POST method.

File: src/tools/common.ts [hunk 4]
```diff
@@ -92,3 +92,19 @@ export function stringVisualLength (str: string): number {
   const split = splitter.splitGraphemes(str)
   return split.length
 }
+
+/**
+ * compute the chainType of the evm chain by coinType.
+ * categorize all evm chains as ETH chainType.
+ * @param coinType
+ */
+export function computedEvmChainTypeByCoinType (coinType: CoinType): ChainType {
```
Comment (zgayjjf) [inline src/tools/common.ts]: Maybe 'compute' is better than `computed`.
And it seems to be not only `evmChainType` but `ChainType`?

File: packages/plugin-register/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      }
+      else {
+        trxHash = await this.signer.sendTransaction({
+          to: orderInfo.receipt_address,
+          value: orderInfo.amount,
+          data: orderInfo.order_id,
+        })
+      }
+
+      await this.bitBuilder.returnTrxHashToService({
```
Comment (zgayjjf) [inline packages/plugin-register/src/index.ts:44]: A `register_result` or something like that should be returned to developers so they can query the registration progress.

**Review by zgayjjf — CHANGES_REQUESTED** [review]

**Comment by zgayjjf** [comment]
The `README` needs to be detailed, clarifying the requirement and other info about each registration method.

File: packages/plugin-register/tests/index.test.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  signer
+})
+const pluginTemplate = new BitPluginRegister()
+dotbit.installPlugin(pluginTemplate)
+
+describe('bitAccount.register()', function () {
+  jest.setTimeout(60 * 1000)
+  it('pay with ETH', async function () {
+    const account = dotbit.account('registeraccounttest009.bit')
+    await account.register({
```
Comment (mozwell) [inline packages/plugin-register/tests/index.test.ts]: We should have a return value of 'register' API and then check the return value in Unit Test

**Review by zgayjjf — APPROVED** [review]

**Pull request merged**